### PR TITLE
Ensure NetworksForRelation never returns a CAAS container address

### PR DESCRIPTION
**Pull request by manadart**
This patch addresses a situation in CAAS models where we call `network-get` in a cross-model relation context and can potentially end up with a container (local-machine) address for ingress if we request a public address that is not yet available.

Now, if the only address is a private one we return nothing for ingress addresses (and accordingly egress subnets).

## QA steps

An authentic real test for this relies on timings out of our control. As a proxy, we can test red/green for the new `TestNetworksForRelationCAASModelCrossModelNoPrivate` test with this change:
```diff
diff --git a/apiserver/facades/agent/uniter/networkinfocaas.go b/apiserver/facades/agent/uniter/networkinfocaas.go
index c66813350a..bfd3e60f6a 100644
--- a/apiserver/facades/agent/uniter/networkinfocaas.go
+++ b/apiserver/facades/agent/uniter/networkinfocaas.go
@@ -137,7 +137,7 @@ func (n *NetworkInfoCAAS) NetworksForRelation(
        }

        if pollAddr {
-               if ingress, err = n.maybeGetUnitAddress(rel, false); err != nil {
+               if ingress, err = n.maybeGetUnitAddress(rel, true); err != nil {
                        return "", nil, nil, errors.Trace(err)
                }
        }
```

## Documentation changes

N/A

## Bug reference

None.

### CODE REVIEW

File: apiserver/facades/agent/uniter/networkinfo_test.go [hunk 1]
```diff
@@ -316,7 +316,7 @@ func (s *networkInfoSuite) TestNetworksForRelationRemoteRelationDelayedPublicAdd
 		return retry.CallArgs{
 			Clock:       clock.WallClock,
 			Delay:       1 * time.Millisecond,
-			MaxDuration: coretesting.LongWait,
+			MaxDuration: coretesting.ShortWait,
```
Comment (manadart) [inline apiserver/facades/agent/uniter/networkinfo_test.go:319]: These shorten the test from 10s.

**Comment by manadart** [comment]
[Do not fallback to private addresses for cross-model CAAS network-get](https://trello.com/c/cEoVxyU3/3198-do-not-fallback-to-private-addresses-for-cross-model-caas-network-get)

**Review by SimonRichardson — APPROVED** [review]
Nice

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**